src/main/webapp/script.js

```diff
@@ -198,57 +96,72 @@ function createCanvasForId(id) {
- * original and blurred images
- * @param {File} imageFile
+ * Updates the page with new uploaded image.
+ * It calls blurring method, puts new and blurred
+ * images on canvases and updates blurRadius input
+ * bar for new image.
+ * @param {String} imageUrl
+ * @param {Array<Rect>} blurAreas
+ * @return {Promise<void>} nothing
  */
-async function putImageOnPageAndBlur(imageFile) {
-  const imageUrl = URL.createObjectURL(imageFile);
-
-  // create Image object from url to put it on canvas.
-  const imageObj = new Image();
-  imageObj.src = imageUrl;
+async function processImage(imageUrl, blurAreas) {
+  const imageObj = await getImageFromUrl(imageUrl);
 
-  // need to wait until image loads to put it anywhere.
-  imageObj.onload = async () => {
-    const inputCanvas = document.getElementById('input-canvas');
-    drawImageOnCanvas(imageObj, inputCanvas);
+  const inputCanvas = document.getElementById('input-canvas');
+  const outputCanvas = document.getElementById('output-canvas');
 
-    const blurAreas = await getBlurAreas(imageFile);
+  // set blurRadiusInput bar max and default values.
+  const blurRadiusInput = document.getElementById('blurring-radius');
 
-    const blurredImage = getImageWithBlurredAreas(blurAreas, imageObj);
+  blurRadiusInput.setAttribute('max', Math.ceil(
+      getAverageRectSize(blurAreas) / (100 * 100) * 12 * 2));
+  blurRadiusInput.setAttribute('value', blurRadiusInput.max / 2);
 
-    const outputCanvas = document.getElementById('output-canvas');
+  // reblur image every time user scrolls the blurRadiusInput bar.
+  blurRadiusInput.addEventListener('change', (event) => {
+    const blurredImage = getImageWithBlurredAreas(
+        blurAreas, imageObj, blurRadiusInput.value);
     drawImageOnCanvas(blurredImage, outputCanvas);
-  };
+  });
+
+  // draw original and blurred images on page.
+  drawImageOnCanvas(imageObj, inputCanvas);
+  const blurredImage = getImageWithBlurredAreas(
+      blurAreas, imageObj, blurRadiusInput.value);
+  drawImageOnCanvas(blurredImage, outputCanvas);
 }
 
 /**
- * Function to draw sample image (blurred and not blurred)
- * on input and output canvases.
+ * Helper function to get average rect size of
+ * rects to blur. Rect size is its width x height.
+ * @param {Array<Rect>} blurAreas
+ * @return {Number} average rect size.
  */
-async function putSampleImagesOnPage() {
-  const SAMPLE_IMAGE_URL = 'images/hadgehog.jpg';
-  const BLURRED_SAMPLE_IMAGE_URL = 'images/blurred-hadgehog.png';
-
-  /**
-   * @param {string} imageUrl
-   * @param {string} canvasId
-   */
-  const putImageOnCanvas = (imageUrl, canvasId) => {
-    // create Image object from url to put it on canvas.
-    const imageObj = new Image();
-    imageObj.src = imageUrl;
-
-    // need to wait until image loads to put it anywhere.
-    imageObj.onload = () => {
-      const canvas = document.getElementById(canvasId);
-      drawImageOnCanvas(imageObj, canvas);
-    };
-  };
+function getAverageRectSize(blurAreas) {
```

Comment:
In my opinion getAverageRectsArea would be a better name for this function. Also the parameter has nothing to do with blurring, so maybe just change blurAreas to rects or rectangles.